crates/common/trie/db.rs

```diff
@@ -1,24 +1,25 @@
-use crate::{NodeHash, error::TrieError};
+use crate::error::TrieError;
+use ethereum_types::H256;
 use std::{
     collections::HashMap,
     sync::{Arc, Mutex},
 };
 
 pub trait TrieDB: Send + Sync {
-    fn get(&self, key: NodeHash) -> Result<Option<Vec<u8>>, TrieError>;
-    fn put_batch(&self, key_values: Vec<(NodeHash, Vec<u8>)>) -> Result<(), TrieError>;
-    fn put(&self, key: NodeHash, value: Vec<u8>) -> Result<(), TrieError> {
+    fn get(&self, key: H256) -> Result<Option<Vec<u8>>, TrieError>;
+    fn put_batch(&self, key_values: Vec<(H256, Vec<u8>)>) -> Result<(), TrieError>;
+    fn put(&self, key: H256, value: Vec<u8>) -> Result<(), TrieError> {
```

Comment:
Maybe we could leave `NodeHash` as a type alias for `H256`? I think it would make the api a bit clearer, as we also use `H256` as keys for state and storage tries and this may cause confusion

Reply:
> Maybe we could leave NodeHash as a type alias for H256?
That's what I did at first, but it won't work because we still need the `NodeHash` enum for serialization purposes.

Having a different type alias would work, but in that case I think it'd be better to have a newtype struct (a wrapper) that implements `Encodeable` and can be used as mbdx table keys (the `H256` cannot be used, that's why there's `[u8; 32]` where there were `NodeHash`es before. Should I try that?